### PR TITLE
Refactor optimizer and lr scheduler part

### PR DESCRIPTION
**Pull request by vinnamkim**
### Summary
 - Ticket no. 135933
 - This PR aims to remove the custom code in `src/otx/cli/cli.py` which is hardly coupled with `OTXModel`.
 - To do this, change type `optimizer` and `scheduler` typings from

    ```python
    optimizer: list[OptimizerCallable] | OptimizerCallable = DefaultOptimizerCallable,
    scheduler: list[LRSchedulerCallable] | LRSchedulerCallable = DefaultSchedulerCallable,
    ```

    to

    ```python
    optimizer: OptimizerCallable = DefaultOptimizerCallable,
    scheduler: LRSchedulerCallable | LRSchedulerListCallable = DefaultSchedulerCallable,
    ```

    This is to remove the `list[OptimizerCallable]` and `list[LRSchedulerCallable]` typings. The reason for this is that I faced this issue, https://github.com/omni-us/jsonargparse/issues/478, after removing `add_list_type_arguments()` from `src/otx/cli/cli.py`.

 - According to this type change, it is required to implement `LinearWarmupSchedulerCallable` newly to incorporate the linear warmup scheduler with the other main scheduler.
 - `LinearWarmupSchedulerCallable` and `OTXModel.lr_scheduler_step()` will solve this problem, https://github.com/openvinotoolkit/training_extensions/pull/3035#discussion_r1515429701

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/algo/classification/mmconfigs/multiclass_classification/dino_v2.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+backbone:
+  name: dinov2_vits14_reg
+  frozen: false
+head:
+  in_channels: 384
+  num_classes: 1000
```
Comment (harimkang) [inline src/otx/algo/classification/mmconfigs/multiclass_classification/dino_v2.yaml:6]: Wouldn't it be better to have this as an argument to the model? DINO doesn't actually use mm, so it seems a bit odd to have configs moved to mmconfigs.

File: src/otx/algo/anomaly/openvino_model.py [hunk 2]
```diff
@@ -12,17 +13,17 @@
 
 from typing import TYPE_CHECKING, Any
 
-from lightning.pytorch import LightningModule
-
+from otx.core.metrics import NullMetricCallable
+from otx.core.metrics.types import MetricCallable
```
Comment (harimkang) [inline src/otx/algo/anomaly/openvino_model.py]:
```suggestion
from otx.core.metrics.types import MetricCallable, NullMetricCallable
```

File: src/otx/engine/utils/auto_configurator.py [hunk 3]
```diff
@@ -256,41 +254,60 @@ def get_model(self, model_name: str | None = None, label_info: LabelInfo | None
...
+            model_config["init_args"]["hlabel_info"] = label_info
+        elif label_info is None and self.task == OTXTaskType.H_LABEL_CLS:
+            msg = "You should explicitly give label_info for `OTXTaskType.H_LABEL_CLS` task."
+            raise ValueError(msg)
 
-        logger.warning(f"Set Default Model: {self.config['model']}")
-        return instantiate_class(args=(), init=self.config["model"])
+        return model_parser.instantiate_classes(Namespace(model=model_config)).get("model")
 
     def get_optimizer(self) -> list[OptimizerCallable] | None:
```
Comment (harimkang) [inline src/otx/engine/utils/auto_configurator.py:278]: This seems to be deprecated in `Engine`, do we still need this function?
The same goes for `get_scheduler`.

File: src/otx/algo/classification/mmconfigs/multiclass_classification/dino_v2.yaml (hunk 1 above)
Comment (vinnamkim) [reply src/otx/algo/classification/mmconfigs/multiclass_classification/dino_v2.yaml:6]: It is currently implemented as MM-style and it must be revisited in the future. The reason for this change is because I encountered annoying test errors where `jsonargparse` kept failing to parse something in `DictConfig`. What you mentioned will be amortized in the future revisit for this model to disassemble the MM-style.

File: src/otx/engine/utils/auto_configurator.py (hunk 3 above)
Comment (vinnamkim) [reply src/otx/engine/utils/auto_configurator.py:278]: If you really think it as what you said, you can remove this. However, I'll keep this because it is a big sized PR, so that I'd like to avoid any potential test error by removing it.

File: src/otx/core/metrics/fmeasure.py [hunk 4]
```diff
@@ -644,6 +644,7 @@ class FMeasure(Metric):
     def __init__(
         self,
         label_info: LabelInfo,
+        *,
```
Comment (jaegukhyun) [inline src/otx/core/metrics/fmeasure.py:647]: Why is this needed?

**Review by jaegukhyun — APPROVED** [review]

File: src/otx/engine/utils/auto_configurator.py (hunk 3 above)
Comment (harimkang) [reply src/otx/engine/utils/auto_configurator.py:278]: Okay, I see, this is something I'll look into and remove later.

**Review by harimkang — APPROVED** [review]

File: src/otx/core/metrics/fmeasure.py (hunk 4 above)
Comment (vinnamkim) [reply src/otx/core/metrics/fmeasure.py:647]: I'd like to make sure that only `label_info` is an unique positional argument: https://realpython.com/python-asterisk-and-slash-special-parameters/#in-short-the-python-asterisk-and-slash-control-how-to-pass-values-to-functions

It does not affect to the runtime behavior.
